src/main/scala/org/apache/spark/sql/delta/storage/BaseExternalLogStore.scala

```diff
@@ -90,23 +103,31 @@ abstract class BaseExternalLogStore(sparkConf: SparkConf, hadoopConf: Configurat
    * Method try to rewrite TransactionLog from temporary path if it not exists.
    * Method returns completed [[LogEntryMetadata]]
    */
-
-   private def tryFixTransactionLog(fs: FileSystem, entry: LogEntryMetadata): LogEntryMetadata = {
+  private def tryFixTransactionLog(fs: FileSystem, entry: LogEntryMetadata): LogEntryMetadata = {
     logDebug(s"Try to fix: ${entry.path}")
-    val tempPath = entry.tempPath.get
     try {
-      writeActions(fs, entry.path, read(tempPath).iterator)
-      val completedEntry = entry.complete()
-      writeCache(fs, completedEntry, overwrite = true)
-      deleteFile(fs, tempPath)
-      completedEntry
+      writeLogTransaction(fs, entry)
+    } catch {
+      case e: FileSystemException => print(
+        s"Failed to fix transaction, ignore... ${e.getMessage}\n"
+      )
+    }
+    val completedEntry = entry.complete()
+    writeCache(fs, completedEntry, overwrite = true)
+    completedEntry
+  }
+
+  private def writeLogTransaction(fs: FileSystem, entryMetadata: LogEntryMetadata) {
```

Comment:
what about renaming to writeCompleteLogTransaction  or completeLogTransaction?

Reply:
right